servicetalk-grpc-protoc/src/main/java/io/servicetalk/grpc/protoc/Generator.java

```diff
@@ -873,14 +877,13 @@ private void extractJavaDocComments(State state, int methodIndex, MethodSpec.Bui
                             printJavaDocs, (methodName, b) -> {
                                 ClassName inClass = messageTypesMap.get(clientMetaData.methodProto.getInputType());
                                 b.addModifiers(ABSTRACT).addParameter(clientMetaData.className, metadata)
-                                .addAnnotation(Deprecated.class)
-                                .addJavadoc(JAVADOC_DEPRECATED + "Use {@link #$L($T,$T)}." + lineSeparator(),
-                                        methodName,
-                                        GrpcClientMetadata, clientMetaData.methodProto.getClientStreaming() ?
-                                                Publisher : inClass);
+                                .addAnnotation(Deprecated.class);
                                 if (printJavaDocs) {
                                     extractJavaDocComments(state, methodIndex, b);
-                                    b.addJavadoc(JAVADOC_PARAM + metadata +
+                                    b.addJavadoc(JAVADOC_DEPRECATED + "Use {@link #$L($T,$T)}." + lineSeparator(),
```

Comment:
I moved this to be conditional, however adding it to the end of this block won't make it after all the params because there are params added after this function is called. we can add additional callback to add deprecated in the right place but I'm not sure it is worth the additional complexity for something we will be removing...